sdk/daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/crypto/HashSpec.scala

```diff
@@ -219,6 +259,46 @@ class HashSpec extends AnyWordSpec with Matchers {
       hash1 should !==(hash2)
     }
 
+    "not produce collision in Map of records" in {
+      def ref4(x4: Long) = {
+        ValueRecord(
+          None,
+          ImmArray(
+            None -> ValueUnit,
+            None -> ValueUnit,
+            None -> ValueUnit,
+            None -> ValueUnit,
+            None -> ValueUnit,
+            None -> ValueUnit,
+            None -> ValueUnit,
+            None -> ValueUnit,
+            None -> ValueInt64(x4),
+          ),
+        )
+      }
+
+      def genmap(elements: (String, Value)*) =
+        ValueGenMap(ImmArray.from(elements.map { case (k, v) => ValueText(k) -> v }))
+      val value1 = genmap(
+        "0" -> // '00000001' + '30'
+          ref4(0x3030303030303030L), // '00000008' + '3030303030303030'
+        "00000000" -> // '00000008' + '3030303030303030'
+          ref4(0), // <empty>
+      )
+      val value2 = genmap(
+        "0" -> // '00000001' + '30'
+          ref4(0), // <empty>
```

Comment:
Is this test still meaningful? I mean, the hashing for `GenMap`s no longer looks at default values, so the hashed string here actually is NOT empty, unlike what the comment says.